### PR TITLE
Reduce compiler warnings related to rlp encode int

### PR DESCRIPTION
**Pull request by jangko**

### CODE REVIEW

File: eth/p2p/rlpx.nim [hunk 1]
```diff
@@ -1000,7 +1000,7 @@ proc p2pProtocolBackendImpl*(protocol: P2PProtocol): Backend =
       var `rlpWriter` = `initRlpWriter`()
       const `perProtocolMsgIdVar` {.used.} = `msgId`
       let `perPeerMsgIdVar` = `perPeerMsgIdValue`
-      `append`(`rlpWriter`, `perPeerMsgIdVar`)
+      `append`(`rlpWriter`, `perPeerMsgIdVar`.uint)
```
Comment (arnetheduck) [inline eth/p2p/rlpx.nim:1003]: this one we should change in the rplx layer - this conversion here is invalid

Comment (arnetheduck) [reply eth/p2p/rlpx.nim:1003]: in fact, we should make it a hard error when it's signed

Comment (jangko) [reply eth/p2p/rlpx.nim:1003]: rlpx need cleanup and major refactoring. will open an issue for the time being.

Comment (arnetheduck) [reply eth/p2p/rlpx.nim:1003]: ok, let's remove this from here for now then - it's better to have the warning in the code until it is fixed - it's a dangerous conversion that can panic, hidden deep in macro code => not godd.

**Pull request merged**